tests/image/image_data_generator_test.py

```diff
@@ -442,5 +442,34 @@ def test_random_transforms():
     assert transform_dict['brightness'] is None
 
 
+def test_fit_rescale(all_test_images):
+    rescale = 1. / 255
+
+    for test_images in all_test_images:
+        img_list = []
+        for im in test_images:
+            img_list.append(utils.img_to_array(im)[None, ...])
+        images = np.vstack(img_list)
+
+        # featurewise_center and featurewise_std_normalization test
+        generator = image_data_generator.ImageDataGenerator(
+            rescale=rescale,
+            featurewise_center=True,
+            featurewise_std_normalization=True)
+        generator.fit(images)
+        batch = generator.flow(images, batch_size=8).next()
+        assert abs(np.mean(batch)) < 1e-3
```

Comment:
please make it a bit tighter 1E-6

Reply:
Even without my changes to the code I couldn’t produce outputs with constant precision higher than 1e-6. Mostly it is in the range of 1e-4 to 1e-7 (tried on both Ubuntu and Windows).

Reply:
that's a bit weird, I would expect that precision level. Any ideas why or from where is the difference coming from?

Reply:
I found out why this happened. The reason is that even if I pass `dtype='float64'` to the `ImageDataGenerator`, it creates an iterator (`NumpyArrayIterator`, `DirectoryIterator`, or `DataFrameIterator`) using the default `dtype='float32'` thus decreasing the precision. I fixed this problem in the latest commit and set precision in tests to 1e-6.